src/Microsoft.ML.CodeGenerator/Utils.cs

```diff
@@ -309,12 +281,45 @@ internal static IList<string> GenerateClassLabels(ColumnInferenceResults columnI
                     result.Add($"[ColumnName(\"{columnName}\"), LoadColumn({column.Source[0].Min})]");
                 }
                 sb.Append(" ");
-                sb.Append(Utils.Normalize(column.Name));
+                string normalizedColumnName = Utils.Normalize(column.Name);
+                if (columnNames.Contains(normalizedColumnName))
+                {
+                    normalizedColumnName = normalizedColumnName + "_" + GetSymbolOfDataKind(dataKind);
+                    if (columnNames.Contains(normalizedColumnName))
+                        throw new ArgumentException($"The column '{column.Name}' with type '{dataKind}' is not unique in the dataset.");
```

Comment:
so it will still throw an error when user has (col:int, col:string and Col:string) in their dataset?

Reply:
Currently yes. Thank you for noting this. When there is an input column of the same name and type and when `columnMapping = null`, it currently throws an error. However, I can implement to do the following: 

When 2 input columns share the same name and type (say 1 "Country" and 2 "country" and `typeof(country)=string' columns, the second identical "country" column could be named as "Country_string_1". Would we like to implement this, or would we want to expect unique columns of the exact name(case-sensitive) and type?

Reply:
Hi @LittleLittleCloud , I just updated my code to address your concern.